airflow/providers/common/sql/hooks/sql.py

```diff
@@ -173,7 +175,17 @@ def __init__(self, *args, schema: str | None = None, log_sql: bool = True, **kwa
         )
 
     @property
-    def placeholder(self) -> str:
+    def placeholder(self):
+        conn = self.get_connection(getattr(self, self.conn_name_attr))
+        placeholder = conn.extra_dejson.get("placeholder")
+        if placeholder in DEFAULT_SQL_PLACEHOLDERS:
+            return placeholder
+        self.log.warning(
+            "Placeholder defined in Connection '%s' is not listed in 'DEFAULT_SQL_PLACEHOLDERS' "
+            "and got ignored. Falling back to the default placeholder '%s'.",
+            placeholder,
```

Comment:
I wonder if this is right approach?
I would prefer to warn when using placeholder that is different with what we know but if user still wish to use something that is different we should allow it and not prevent it.

Reply:
We can change it if you want, but that's how it was originally implemented in the OdbcHook.